graylog2-server/src/main/java/org/graylog2/indexer/messages/Messages.java

```diff
@@ -164,210 +75,47 @@ public ResultMessage get(String messageId, String index) throws DocumentNotFound
-        List<BulkResult.BulkResultItem> failedItems = new ArrayList<>();
-        for (;;) {
-            try {
-                List<BulkResult.BulkResultItem> failures = bulkIndexChunked(messageList, isSystemTraffic, offset, chunkSize);
-                failedItems.addAll(failures);
-                break; // on success
-            } catch (EntityTooLargeException e) {
-                LOG.warn("Bulk index failed with 'Request Entity Too Large' error. Retrying by splitting up batch size <{}>.", chunkSize);
-                if (chunkSize == messageList.size()) {
-                    LOG.warn("Consider lowering the \"output_batch_size\" setting.");
-                }
-                failedItems.addAll(e.failedItems);
-                offset += e.indexedSuccessfully;
-                chunkSize /= 2;
-            }
-            if (chunkSize == 0) {
-                throw new ElasticsearchException("Bulk index cannot split output batch any further.");
-            }
-        }
-
-        if (!failedItems.isEmpty()) {
-            final Set<String> failedIds = failedItems.stream().map(item -> item.id).collect(Collectors.toSet());
-            recordTimestamp(messageList, failedIds);
-            return propagateFailure(failedItems, messageList);
-        } else {
-            recordTimestamp(messageList, Collections.emptySet());
-            return Collections.emptyList();
-        }
-    }
-
-    private List<BulkResult.BulkResultItem> bulkIndexChunked(final List<Map.Entry<IndexSet, Message>> messageList, boolean isSystemTraffic, int offset, int chunkSize) throws EntityTooLargeException {
-        chunkSize = Math.min(messageList.size(), chunkSize);
-
-        final List<BulkResult.BulkResultItem> failedItems = new ArrayList<>();
-        final Iterable<List<Map.Entry<IndexSet, Message>>> chunks = Iterables.partition(messageList.subList(offset, messageList.size()), chunkSize);
-        int chunkCount = 1;
-        int indexedSuccessfully = 0;
-        for (List<Map.Entry<IndexSet, Message>> chunk : chunks) {
-
-            long messageSizes = chunk.stream().mapToLong(m -> m.getValue().getSize()).sum();
-
-            final BulkResult result = bulkIndexChunk(chunk);
-
-            if (result.getResponseCode() == 413) {
-                throw new EntityTooLargeException(indexedSuccessfully, failedItems);
-            }
+        final List<IndexingRequest> indexingRequestList = messageList.stream()
+                .map(entry -> IndexingRequest.create(entry.getKey(), entry.getValue()))
+                .collect(Collectors.toList());
 
-            // TODO should we check result.isSucceeded()?
+        final List<IndexFailure> indexFailures = messagesAdapter.bulkIndex(indexingRequestList);
 
-            indexedSuccessfully += chunk.size();
+        final Set<String> failedIds = indexFailures.stream().map(IndexFailure::letterId).collect(Collectors.toSet());
+        final List<IndexingRequest> successfullRequests = indexingRequestList.stream()
```

Comment:
IntelliJ says this is a typo and should be renamed `successfulRequests` with one `l`. I believe it blindly!